standalone/src/models/shipping-method/shipping-rate/builders.spec.ts

```diff
@@ -0,0 +1,96 @@
+import type { TShippingRateRest, TShippingRateGraphql } from './types';
+import { ShippingRateRest, ShippingRateGraphql, ShippingRate } from './index';
+
+function validateRestModel(model: TShippingRateRest) {
+  expect(model).toEqual(
+    expect.objectContaining({
+      freeAbove: expect.objectContaining({
+        centAmount: expect.any(Number),
+        currencyCode: expect.any(String),
+      }),
+      price: expect.objectContaining({
+        type: 'centPrecision',
+        centAmount: expect.any(Number),
+        currencyCode: expect.any(String),
+      }),
+      tiers: [
+        expect.objectContaining({
+          isMatching: expect.any(Boolean),
+          price: expect.objectContaining({
+            centAmount: expect.any(Number),
+            currencyCode: expect.any(String),
+          }),
+          score: expect.any(Number),
+          priceFunction: null,
+          type: 'CartScore',
+        }),
+      ],
+      isMatching: expect.any(Boolean),
+    })
+  );
+}
+
+function validateGraphqlModel(model: TShippingRateGraphql) {
```

Comment:
suggestion: We could consolidate just one validation method and check the `__typename` only in the Graphql tests.